WireDomain/Sources/WireDomain/Event Processing/ConversationEventProcessor/ConversationMLSResetEventProcessor.swift

```diff
@@ -24,65 +24,85 @@ struct ConversationMLSResetEventProcessor: ConversationMLSResetEventProcessorPro
 
     enum Failure: Error {
         case conversationNotFound
-        case invalidArguments
-        case failedToWipeMLSConversation
+        case failedToGetMLSGroupIDs
     }
 
     private let mlsService: any MLSServiceInterface
     private let conversationLocalStore: any ConversationLocalStoreProtocol
-    private let featureRepository: any FeatureRepositoryInterface
+    private let featureConfigRepository: any FeatureConfigRepositoryProtocol
+    private let userDefaultsRepository: ResetMLSConversationUserDefaultsRepositoryProtocol
 
     init(
         mlsService: any MLSServiceInterface,
         conversationLocalStore: any ConversationLocalStoreProtocol,
-        featureRepository: any FeatureRepositoryInterface
+        featureConfigRepository: any FeatureConfigRepositoryProtocol,
+        userDefaultsRepository: ResetMLSConversationUserDefaultsRepositoryProtocol
     ) {
         self.mlsService = mlsService
         self.conversationLocalStore = conversationLocalStore
-        self.featureRepository = featureRepository
+        self.featureConfigRepository = featureConfigRepository
+        self.userDefaultsRepository = userDefaultsRepository
     }
 
     func processEvent(_ event: ConversationMLSResetEvent) async throws {
+        do {
+            let attributes: LogAttributes = [.conversationId: event.conversationID.id.safeForLoggingDescription]
 
-        let feature = await featureRepository.fetchAllowedGlobalOperations()
-        guard feature.status == .enabled, feature.config.mlsConversationReset == true else {
-            WireLogger.mls.debug(
-                "No need to process reset broken MLS conversation, FF is OFF"
-            )
-            return
-        }
+            let feature = try await featureConfigRepository.fetchAllowedGlobalOperations()
+            guard feature.status == .enabled, feature.config?.mlsConversationReset == true else {
+                WireLogger.mls.debug(
+                    "No need to process reset broken MLS conversation, FF is OFF",
+                    attributes: attributes
+                )
+                return
+            }
```

Comment:
todo: please double check this check is necessary. From the useCase i understood this should always happen.
https://wearezeta.atlassian.net/wiki/spaces/ENGINEERIN/pages/1997013074/Use+case+Reset+fix+a+broken+MLS+conversation#Conversation-reset-event